scripts/cache_mulled_singularity_containers.py

```diff
@@ -0,0 +1,71 @@
+from bioblend.galaxy import GalaxyInstance
+import argparse
+import os
+import logging
+import time
+
+def get_args():
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument('-a', '--api-key',
+                            required=True,
+                            help='Admin API Key')
+    arg_parser.add_argument('-u', '--url',
+                            required=True,
+                            help='Galaxy URL')
+    arg_parser.add_argument('--debug',
+                            action='store_true',
+                            default=False,
+                            help='Print debug information')
+    args = arg_parser.parse_args()
+    return args
+
+def set_logging_level(debug=False):
+    logging.basicConfig(
+        level=logging.DEBUG if debug else logging.INFO,
+        format='%(asctime)s - %(message)s',
+        datefmt='%d-%m-%y %H:%M:%S')
+
+def main():
+    """
+    Downloads singularity containers on a Galaxy instance which is configured
+    to use such dependency resolvers (mulles_singularity and cached_mulled_singularity).
+
+    It will request the resolution of tools to containers from the instance,
+    and then for all mulled_singularity resolved tools that are pointing to
+    docker://<container-url>, it will download and build the singularity SIF.
+    This download is done only once per container, and applies after for all tools
+    that will make use of it.
+    """
+    args = get_args()
+    set_logging_level(debug=args.debug)
+    gi = GalaxyInstance(url=args.url, key=arg.key)
```

Comment:
@pcm32 I'm not sure where the `arg.key` is coming from here? Is that supposed to be `args.api_key` ?

Reply:
Well spotted! I was doing some changes that I shouldn't have done after testing.
```suggestion
    gi = GalaxyInstance(url=args.url, key=args.key)
```